judge/views/contests.py

```diff
@@ -217,6 +217,11 @@ def get_context_data(self, **kwargs):
             .annotate(has_public_editorial=Sum(Case(When(solution__is_public=True, then=1),
                                                     default=0, output_field=IntegerField()))) \
             .add_i18n_name(self.request.LANGUAGE_CODE)
+        context['contest_has_public_editorials'] = False
+        for contest_problem in context['contest_problems']:
```

Comment:
This sounds like a job for `any`, e.g. `context['contest_has_public_editorials'] = any(problem.has_public_editorial for problem in context['contest_problems']`.

Though, doesn't this check have to also include whether the problem is public? L144 in contest.html checks that too.